ports-of-call/robust_utils.hpp

```diff
@@ -64,13 +64,40 @@ PORTABLE_FORCEINLINE_FUNCTION auto ratio(const A &a, const B &b) {
   return a / (b + sgn(b) * SMALL<B>());
 }
 
+template <typename DataType>
+PORTABLE_FUNCTION constexpr DataType reldiff(const DataType actual, const DataType expected)
+{
+    if (actual == expected) {
+        return 0;
+    } else if (expected == 0) {
+        // expected 0, got nonzero, define this to be 100% error
+        return 1;
+    } else {
+        // Note: we do not use std::abs to calculate the absolute value because
+        //       it is not constexpr until C++23, therefore it is not portable
+        //       to GPUs (currently using C++17).
+        const auto signed_reldiff = (actual - expected) / expected;
+        return (signed_reldiff >= 0 ? signed_reldiff : -signed_reldiff);
+    }
+}
+
```

Comment:
A simple relative difference calculation that is used extensively in the Singe unit tests and seemed a good candidate for here given that other codes probably need something similar for their unit tests.  If you feel this is out of scope, please raise an objection.